src/LinkedList.ts

```diff
@@ -2,54 +2,52 @@ import { fabric } from "fabric";
 import { Node } from "./Node";
 import { makeLine } from "./Utils";
 import Config = require("./Config");
+import { Variable } from "./Variable";
 
 export class LinkedList {
 
-    private head: Node = null;
+    private globalVars: { [key: string]: Variable };
+    private nodes: Node[];
     private canvas: fabric.Canvas;
 
-    constructor(canvasEl: HTMLCanvasElement, values:number[]=[]) {
+    constructor(canvasEl: HTMLCanvasElement, values: number[] = []) {
 
         this.canvas = new fabric.Canvas(canvasEl);
+        this.canvas.selection = false;
 
-        values = values.reverse();
+        // this allows us to draw circles using the coordinates of their centers
+        fabric.Object.prototype.originX = fabric.Object.prototype.originY = 'center';
+
+        this.nodes = [];
         for (let num of values) {
-            this.head = new Node(num, this.head);
+            this.nodes.push(new Node(num, this.canvas));
         }
 
-        this.draw()
-        
-        let nul = new fabric.IText('NULL', {
-            left: Config.LIST_X + (values.length * Config.NODE_SPACE),
-            top: Config.LIST_Y + (Config.NODE_SIZE / 2),
-            fill: '#black',
-        });
-        this.canvas.add(nul) 
+        for (let i = 0; i < this.nodes.length - 1; ++i) {
+            this.nodes[i].next.destination = this.nodes[i + 1];
+        }
+
+        this.globalVars = {
+            head: new Variable("head", this.canvas),
+        }
+        this.globalVars.head.value.destination = this.nodes[0];
```

Comment:
this seems like some breaking of encapsulation here, accessing this many levels of properties, some of which really should be private. Could we have the thing that should be pointed to passed as an argument to the Variable constructor or something?